mustache.js

```diff
@@ -420,6 +426,19 @@
           }
         } else {
           intermediateValue = context.view[name];
+
+          /**
+           * Only checking against hasProperty, which always returns false if
+           * context.view is not an object. Deliberately omitting the check
+           * against primitiveHasOwnProperty if dot notation is not used.
+           *
+           * Consider this example:
+           * Mustache.render("{{#length}}{{length}}{{/length}}", {length: "hello"})
```

Comment:
I was thinking of something like:

```js
Mustache.render(
  "{{#sections}}{{name.length}}: {{quantity}}/{{length}}\n{{/sections}}",
  {
    length: 42,
    sections: [
      { name: "red",   quantity:  7 },
      { name: "green", quantity: 23 },
      { name: "blue",  quantity: 12 }
    ]
  }
)
```

or similar. It be nicer if it was simpler / shorter.